client/starwhale/core/dataset/type.py

```diff
@@ -702,3 +702,53 @@ def create_by_yaml(cls, fpath: t.Union[str, Path]) -> DatasetConfig:
         c = load_yaml(fpath)
 
         return cls(**c)
+
+
+class JsonDict(SwObject):
+    """
+    JsonDict takes json like dict(https://www.json.org/json-en.html) as init parameter
+    Besides the standard value types, SwObject is an extra value type that is allowed to be passed in
+    """
+
+    def __init__(self, d: dict = {}) -> None:
+        for _k, _v in d.items():
+            if type(_k) != str:
```

Comment:
why the key type must be str type?

```python
In [1]: a = {1: "test"}

In [2]: type(a)
Out[2]: dict
```

Reply:
Technically, this is not a full python dict support. As the PR's title indicates, this is a subset of python dict which could be called `JSON like` dict. Keys of JSON should be str type. 
From user's aspect, I can think of little cases where non-str keys are used.
![image](https://www.json.org/img/object.png)

Reply:
ok